replicator/src/main.rs

```diff
@@ -9,20 +11,32 @@ mod config;
 mod core;
 mod migrations;
 
-#[tokio::main]
-async fn main() -> anyhow::Result<()> {
+fn main() -> anyhow::Result<()> {
     let app_name = env!("CARGO_BIN_NAME");
 
-    // Initialize Sentry with replicator-specific tag
-    let _sentry_guard = init_sentry()?;
-
     // We pass `emit_on_span_close = false` to avoid emitting logs on span close
     // for replicator because it is not a web server, and we don't need to emit logs
     // for every closing span.
     let _log_flusher = init_tracing(app_name, false)?;
 
+    // Initialize Sentry before the async runtime starts
+    let _sentry_guard = init_sentry()?;
+
+    // We start the runtime.
+    tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()?
+        .block_on(async_main())?;
+
+    Ok(())
+}
+
+async fn async_main() -> anyhow::Result<()> {
     // We start the replicator.
-    start_replicator().await?;
+    if let Err(err) = start_replicator().await {
+        sentry::capture_error(err.as_dyn_error());
```

Comment:
Also send to logflare?

Reply:
This is a separate system from LogFlare, the event is just sent to sentry, the error is propagated so that it works as before.

Reply:
What I meant was to just add a `error!("{}", err);` which would have sent it to logflare as an error log.

Reply:
Ah right, I can do this!